packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -641,6 +657,28 @@ function propagateParentContextChanges(
   workInProgress.flags |= DidPropagateContext;
 }
 
+function checkIfComparedContextValuesChanged(
+  oldComparedValue: mixed,
+  newComparedValue: mixed,
+): boolean {
+  if (isArray(oldComparedValue) && isArray(newComparedValue)) {
+    for (
+      let i = 0;
+      i < oldComparedValue.length && i < newComparedValue.length;
+      i++
+    ) {
+      if (!is(newComparedValue[i], oldComparedValue[i])) {
+        return true;
+      }
+    }
+  } else {
```

Comment:
The original selector proposal did not do index based comparison because it's not clear if the returned array is meant to represent a multi-selection or it just happens that the selected value was an array. It does mean that returning complex objects will void the comparison benefit a bit. Since this is a compiler target you can I suppose just force that the return value is always an array that way you can safely compare inner array values on their identity

Reply:
Good callout. I mostly wanted to address multiple properties being destructured from one context access, but we do need to handle the case of a single used value being an array. And we wouldn't want to compare each value in since it is potentially a long list. So for now we'll plan to begin the test by always wrapping returns in array. Will leave the fall through to `is()` just to have the option value to update compiler output without changing the runtime if we want to test other return types from compare functions.